examples/file-uploader/js/app.js

```diff
@@ -0,0 +1,75 @@
+
+const Type = require('union-type');
+const T = require('ramda/src/T')
+    , assoc = require('ramda/src/assoc')
+    , curry  = require('ramda/src/curry')
+    , compose  = require('ramda/src/compose')
+    , map  = require('ramda/src/map')
+    , invoker = require('ramda/src/invoker') 
+;
+const h = require('snabbdom/h');
+
+const uploadList = require('./list');
+const uploader   = require('./uploader');
+  
+
+// action
+
+const listUpdate = (listAction,model) => {
+  const [state, tasks] = uploadList.update(listAction, model.uploads);
+  return [ assoc('uploads', state, model), 
+           tasks.map( map(Action.Route) ) 
+         ];
+}
+
+const Action = Type({
+  Create: [T, T],
+  Route:  [uploadList.Action]
+});
+
+const update = Action.caseOn({
+  Create: (up,files,model) => {
+    return listUpdate( uploadList.Action.Create(up,files), model );
+  },
+
+  Route: listUpdate
+});
+
+
+// model
+
+const init = () => { return { uploads: uploadList.init() }; }
+
+// view
+
+const view = curry( ({url, headers, action$}, model) => {
+  
+  const up = uploader.upload(headers, url);
+
+  const form = (
```

Comment:
Maybe making `form` a separate function would be nice. I think it would make it easier to see what values it depends on (only `update` as far as I can tell).